editor/src/components/canvas/dom-lookup.ts

```diff
@@ -31,35 +31,51 @@ export function findParentSceneValidPaths(target: Element): Array<ElementPath> |
   }
 }
 
+// Take a DOM element, and try to find the nearest selectable path for it
 export function findFirstParentWithValidElementPath(
-  validDynamicElementPathsForLookup: Array<ElementPath> | 'no-filter',
+  validDynamicElementPathsForLookup: Set<string> | 'no-filter',
   target: Element,
 ): ElementPath | null {
   const dynamicElementPaths = getPathsOnDomElement(target)
-  const validStaticElementPathsForScene = findParentSceneValidPaths(target) ?? []
+  const staticAndDynamicTargetElementPaths = dynamicElementPaths.map((p) => {
+    return {
+      static: EP.toString(EP.makeLastPartOfPathStatic(p)),
+      dynamic: p,
+    }
+  })
+  const validStaticElementPathsForSceneArray =
+    findParentSceneValidPaths(target)?.map(EP.toString) ?? []
+  const validStaticElementPathsForScene = new Set(validStaticElementPathsForSceneArray)
+
   const validStaticElementPaths =
     validDynamicElementPathsForLookup === 'no-filter'
       ? validStaticElementPathsForScene
-      : intersection(
-          validDynamicElementPathsForLookup.map(EP.makeLastPartOfPathStatic),
-          validStaticElementPathsForScene,
-          EP.pathsEqual,
+      : new Set(
+          [...validDynamicElementPathsForLookup].filter((p) =>
+            validStaticElementPathsForScene.has(p),
+          ),
         )
 
-  const filteredValidPathsMappedToDynamic = mapDropNulls((validPath: ElementPath) => {
-    return dynamicElementPaths.find((tp) => {
-      const elementPathWithStaticElementPart = EP.makeLastPartOfPathStatic(tp)
-      return EP.pathsEqual(validPath, elementPathWithStaticElementPart)
-    })
-  }, validStaticElementPaths)
+  const filteredValidPathsMappedToDynamic = mapDropNulls(
+    (validPath: string) => {
+      return staticAndDynamicTargetElementPaths.find(
+        (staticAndDynamic) => staticAndDynamic.static === validPath,
+      )?.dynamic
+    },
+    [...validStaticElementPaths],
+  )
 
   if (filteredValidPathsMappedToDynamic.length > 0) {
-    return last(filteredValidPathsMappedToDynamic) ?? null
+    const asPaths = filteredValidPathsMappedToDynamic.sort((l, r) => EP.depth(l) - EP.depth(r))
```

Comment:
maybe sortedSomethingSomething is a better name

Reply:
Done :+1: